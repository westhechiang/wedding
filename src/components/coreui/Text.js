import styled from 'styled-components';
import { theme } from '../../ui';

const hoverEffects = {
  boldFont: `&:hover{ font-weight: ${theme.font.weight.heavy}; }`,
};

const Text = styled.p`
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: ${props => theme.color[props.color]};
  font-size: ${props => theme.font.size[props.size] || theme.font.size.default};
  font-family: ${theme.font.family};
  font-weight: ${props => theme.font.weight[props.weight] || 'normal'};
  line-height: ${props => props.lineHeight || theme.font[props.size]};
  text-indent: ${props => props.textIndent};
  text-align: ${props => (props.textAlign || 'start')};
  text-transform: ${props => (props.textTransform || 'none')};
  margin: ${props => (props.margin || 0)};
  letter-spacing: ${props => props.letterSpacing || 'normal'};
  position: ${props => (props.position || 'relative')};
  top: ${props => (props.top || 'auto')};
  padding: ${props => (props.padding || '0')};
  white-space: ${props => props.whiteSpace || 'normal'};
  width: ${props => props.width || 'auto'};
  flex-wrap: ${props => props.flexWrap || 'auto'};
  display: ${props => props.display || 'inline'};
  opacity: ${props => props.opacity || '1'};
  ${props => hoverEffects[props.hover]}
`;

export default Text;
