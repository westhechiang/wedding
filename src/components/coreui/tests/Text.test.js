import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Text from '../../../components/coreui/Text';

describe('<Text />', () => {
  it('should match its empty snapshot', () => {
    const wrapper = shallow(<Text />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "large" size snapshot', () => {
    const wrapper = shallow(<Text size="large" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "base" size snapshot', () => {
    const wrapper = shallow(<Text size="base" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "small" size snapshot', () => {
    const wrapper = shallow(<Text size="small" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "xsmall" size snapshot', () => {
    const wrapper = shallow(<Text size="xsmall" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "xxsmall" size snapshot', () => {
    const wrapper = shallow(<Text size="xxsmall" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "xxxsmall" size snapshot', () => {
    const wrapper = shallow(<Text size="xxxsmall" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "light" weight snapshot', () => {
    const wrapper = shallow(<Text weight="light" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "semiBold" weight snapshot', () => {
    const wrapper = shallow(<Text weight="semiBold" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "boldFont" hover effect snapshot', () => {
    const wrapper = shallow(<Text hover="boldFont" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
