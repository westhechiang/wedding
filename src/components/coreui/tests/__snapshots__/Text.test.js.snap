// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should match its "base" size snapshot 1`] = `
<p
  className="sc-bdVaJa gnUxiO"
  size="base"
/>
`;

exports[`<Text /> should match its "boldFont" hover effect snapshot 1`] = `
<p
  className="sc-bdVaJa iYihzc"
/>
`;

exports[`<Text /> should match its "large" size snapshot 1`] = `
<p
  className="sc-bdVaJa MNbMg"
  size="large"
/>
`;

exports[`<Text /> should match its "light" weight snapshot 1`] = `
<p
  className="sc-bdVaJa fLOxNH"
/>
`;

exports[`<Text /> should match its "semiBold" weight snapshot 1`] = `
<p
  className="sc-bdVaJa heNjIU"
/>
`;

exports[`<Text /> should match its "small" size snapshot 1`] = `
<p
  className="sc-bdVaJa kFRbqC"
  size="small"
/>
`;

exports[`<Text /> should match its "xsmall" size snapshot 1`] = `
<p
  className="sc-bdVaJa brBLTk"
  size="xsmall"
/>
`;

exports[`<Text /> should match its "xxsmall" size snapshot 1`] = `
<p
  className="sc-bdVaJa iPuzfH"
  size="xxsmall"
/>
`;

exports[`<Text /> should match its "xxxsmall" size snapshot 1`] = `
<p
  className="sc-bdVaJa kTenKW"
  size="xxxsmall"
/>
`;

exports[`<Text /> should match its empty snapshot 1`] = `
<p
  className="sc-bdVaJa iVRnQM"
/>
`;
